docs/calculated-questions.md

```diff
@@ -1,50 +1,50 @@
 
-This feature makes heavy use of Django signals. There are two "main loops" for this.
+The results of calculated questions are of course stored in the corresponding
+answers.  As it is highly dependent on the JEXL expression, what kind
+dependencies to other questions are created and where in the form (sub-forms,
+tables etc.) they are located, it is hard to track when an update of the answer
+is needed.  In our generalistic approach, this complexity is mitigated by
+evaluating greedly at every possible change.
 
+This leads to behavior that might be unexpected compared to other question
+types.
 
-#### `calcDependents`
+First of all, of course a calculated answer is read-only for the GraphQL (and
+python) API.  The answer is fetchable, but there is no
+`CreateDocumentCalculated...Answer`.
 
-Every question that is referenced in a `calcExpression` will memoize the referencing calculated question.
-This list of calculation dependents is mutated whenever a calculated question is created, updated or deleted.
+Second, because the evaluation is greedy, we calculated the answer as soon as
+possible, even if referenced answers are missing.  If this is the case, the
+answer is still created but will have an empty value.
 
-#### Calculated Answers
-
-The results of calculated questions are of course stored in the corresponding answers.
-As it is highly dependent on JEXL expression, what kind dependencies to other questions are created and where in the form (sub-forms, tables etc.) they are located.
-In our generalistic approach, this complexity is mitigated by evaluating greedly at every possible change.
-
-This leads to behavior that might be unexpected compared to other question types.
-
-First of all, of course a calculated answer is read-only for the GraphQL (and python) API.
-The answer is fetchable, but there is no `CreateDocumentCalculated...Answer`.
-
-A couple of mutation lead to the creation or update of a calculated answer.
+A couple of mutations act as entrypoints to evaluation of the export and lead
+to the creation or update of a calculated answer.
 The answer is modified when:
 * the calculated question is modified
-* the calculated question is added to / removed from a form.
+* the calculated question is added to / removed from a form
 * any referenced answer is modified
 * a new document for a form containing calculated questions is modified
 * a new row document of a row form containing calculated questions is created / deleted
 
 
 ### Limitations
 
-The current approach has some limitations that stem from compromises in the calculation of answers.
-
-* 1. calculated questions cannot reference other calculated questions
-* 2. calculated questions can only reference questions that are on the same form-level or lower
+Currently the calculated questions cannot reference other calculated questions. This was an intentional limitation, as it would required additional validation to ensure no circular dependencies are involved.
```

Comment:
This is an intentional limitation to avoid implementational complexity (dependency calculation "propagation", circular dependency checks).

Reply:
Done